README.md

```diff
@@ -1,3 +1,42 @@
 # timeline
 
-Provides a container type `Timeline a` for handling data that changes over time.
+## Motivation
+
+The world is always changing, and often we want to manage the changes of data
+using computers. Below are some concrete examples:
+
+- Employee data such as compensation, city, tax rule, time-off, etc.
+- Prices of products. A product could have different prices on Amazon and EBay,
+  and in different currencies.
+
+Timeline data is often implemented by attaching a field `effective_from` and
+possibly an `effective_to` field to the record. However, only representing and
+storing the data is not sufficient, we need to run operations on timeline data,
```

Comment:
I don't think we should use the word `effective` in this package — the data contained won't necessarily be effective.